sockeye/transformer.py

```diff
@@ -291,14 +293,15 @@ def __init__(self, max_length: int) -> None:
     def forward(self, is_train, req, in_data, out_data, aux):
         # lengths: (batch_size,)
         lengths = in_data[0]
+        dtype = lengths.dtype
 
         # (batch_size, max_length)
-        data = mx.nd.zeros((lengths.shape[0], self.max_length), ctx=lengths.context)
+        data = mx.nd.zeros((lengths.shape[0], self.max_length), dtype=dtype, ctx=lengths.context)
         data = mx.nd.SequenceMask(data=data,
```

Comment:
the bias tensor is added to the data in the encoder. Are you sure we are not causing any fp16 overflow AFTER this addition?

Reply:
actually, this is just a fix for the custom operator. Due to the zeros symbol without dtype it was forced always to be fp32. But in the change, bias calculated in fp32 and casted to fp16 before addition.

Reply:
I am not sure I get what you are saying. I understand that the custom op now creates data in the correct dtype. The problem is that you are masking this fp16 data in some places with the smallest possible value in the fp16 range. Now when you actually use this bias tensor in the transformer model, it gets added to the encoder representation tensors. Depending on how they look like, this addition might cause an overflow: `fp16.min + -x`. 
So instead of using a value for masking that is right at the limit of the representable numeric range, it might be better to use something less extreme.

Reply:
Changed `float16` constant to be something close to maximum, but not maximum possible number.